### ISSUE TITLE
Inject @Param not working

### ISSUE THREAD

**Issue by huksley** [omnifaces/omnifaces#172]
Hi!

I am experiencing problems with  @Param annotation.
I am using Tomcat 8.0.26 + myfaces 2.2.8 + omnifaces-2.1 + primefaces-5.2.jar + weld 2.3.0

In @RequestScoped bean i have
<code java>
@Inject @Param private String someId;
</code>

but it does not get updated on commandLink click or o:commandScript.. invocation.

**Comment by huksley** [comment]
Here is test complete project 
https://github.com/huksley/omnifaces-support/tree/master/TestParamInjection

**Comment by arjantijms** [comment]
The problem seems to be that you're using AJAX to submit the form. f:param only becomes a request (query) parameter for full page requests, and so "@Param" only works for full page requests as well.

**Comment by BalusC** [comment]
Works for me. Your test XHTML is broken. It's missing `<h:head>`, which in turn causes JavaScript errors (press F12 in browser to see it).

**Comment by huksley** [comment]
Sorry for bugs. I fixed it. Added source code and Eclipse project. However it does not work.

AS I understand the code in test must do 3 things
on clicking Test SomeID
1) update SampleBean.someId variable
2) update SomeID: INITED to SomeID: someValue in browser
3) print in stdout
someId = someValue, IN REQUEST someValue2

However, the value in browser stays the same
and in stdout
someId = INITED, IN REQUEST someValue2

P.S. IN REQUEST ... shows value in from ExternalContext getRequestParameterMap
